core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2081,9 +2096,7 @@ class Log(@volatile var dir: File,
         producerStateManager.truncate()
         producerStateManager.updateMapEndOffset(newOffset)
         maybeIncrementFirstUnstableOffset()
-
-        this.recoveryPoint = math.min(newOffset, this.recoveryPoint)
-        this.logStartOffset = newOffset
+        updateLogStartOffset(newOffset)
```

Comment:
Previous behavior is ```recoveryPoint = math.min(newOffset, recoveryPoint))``` but this patch changes it to
```
if (this.recoveryPoint < offset) {
  this.recoveryPoint = offset
}
```
which is equal to ```recoveryPoint = math.max(newOffset, recoveryPoint))```. Is it a bug?

Reply:
I have also updated `updateLogEndOffset` to set the recovery point. In `truncateFully` where we delete all segments and set the log start offset to be equal to the log end offset, this ensures recovery point is also set consistently.

Reply:
Thanks for the explanation!